### PR TITLE
Add channel filter dropdown vue component

### PR DESCRIPTION
**Pull request by bambamboole**
This PR adds a small Vue Component that adds a dropdown filter component for channels to the nav bar.

### CODE REVIEW

File: resources/assets/js/components/ChannelDropdown.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<template>
+    <li class="dropdown">
+        <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true"
+           aria-expanded="false">Channels <span class="caret"></span></a>
+
+        <div class="dropdown-menu" >
```
Comment (concept-core) [inline resources/assets/js/components/ChannelDropdown.vue]: There is some spacing behind `class` which is not needed.

File: resources/assets/js/components/ChannelDropdown.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            <div style="padding:.5rem 1rem">
```
Comment (concept-core) [inline resources/assets/js/components/ChannelDropdown.vue]: I would suggest moving the inline styles in a style tag.

File: resources/assets/js/components/ChannelDropdown.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                <input type="text" class="form-control" v-model="filter" placeholder="Filter Channels..."/>
+            </div>
+            <ul style="max-height: 400px; overflow:auto; ">
+                <li v-for="channel in filteredThreads"><a :href="'/threads/' + channel.slug" v-text="channel.name"></a>
```
Comment (concept-core) [inline resources/assets/js/components/ChannelDropdown.vue]: place `a` tag on new line.

File: resources/assets/js/components/ChannelDropdown.vue [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                </li>
+            </ul>
+        </div>
+    </li>
+</template>
+
+<script>
+    export default {
+        props: ['channels'],
+        data() {
```
Comment (concept-core) [inline resources/assets/js/components/ChannelDropdown.vue]: To stay consistent with other components I would add a new line after the `props` array and `},` of `data()`

File: resources/views/layouts/nav.blade.php [hunk 5]
```diff
@@ -40,16 +40,7 @@
...
-                    <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true"
-                       aria-expanded="false">Channels <span class="caret"></span></a>
-
-                    <ul class="dropdown-menu">
-                        @foreach ($channels as $channel)
-                            <li><a href="/threads/{{ $channel->slug }}">{{ $channel->name }}</a></li>
-                        @endforeach
-                    </ul>
-                </li>
+                <channel-dropdown :channels="{{$channels}}"></channel-dropdown>
```
Comment (concept-core) [inline resources/views/layouts/nav.blade.php]: In the blade views spacing is added for the variables. ( `{{ $channels }}` )

**Comment by bambamboole** [comment]
@concept-core I "fixed" the stuff :-)

Also add the functionality to open the dropdown with the component.

**Comment by concept-core** [comment]
The changes look good but you did break the built.
You might wanna check why and fix it 🙂

**Comment by bambamboole** [comment]
Its issue #114 ...
You can see it in the details.

**Comment by JeffreyWay** [comment]
Thank you!

**Comment by fshequin** [comment]
Why would I be getting this error after incorporating this PR Commit into my codebase:

`[Vue warn]: Error in render: "TypeError: this.channels.filter is not a function"

found in

---> <ChannelDropdown> at resources/assets/js/components/ChannelDropdown.vue
       <Root>`

**Pull request merged**